File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/1878-CheckIfArrayIsSortedAndRotated/1878-CheckIfArrayIsSortedAndRotated.cpp
// Last updated: 8/4/2026, 12:42:08 PM
class Solution {
public:
    bool check(vector<int>& nums) {
        int cnt = 0;
        int n = nums.size();

        if(nums[0] < nums[n-1]) cnt++;

        for(int i = 0 ; i < n-1 ; i++){
            if(nums[i] > nums[i+1]){
                cnt++;
            }
        }
        if(cnt <= 1) return true;
        else return false;
    }
};

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/1878-CheckIfArrayIsSortedAndRotated/1878-CheckIfArrayIsSortedAndRotated.java
// Last updated: 8/4/2026, 12:42:14 PM
class Solution {
    public boolean check(int[] nums) {
        int count = 0;
        for(int i = 0 ; i < nums.length-1 ; i++){
            if(nums[i] > nums[i+1]) count++;
    }
    if(nums[nums.length - 1] > nums[0]) count++;

    if(count <= 1) return true;
    return false;

    }
}